user-interface/src/main/java/life/qbic/datamanager/views/general/MultiSelectLazyLoadingGrid.java

```diff
@@ -51,16 +51,17 @@ private Checkbox renderCheckbox(T bean) {
 
       if (event.isFromClient()) {
         Boolean allBoxVal = selectAllCheckBox.getValue();
-        if(Boolean.TRUE.equals(allBoxVal) && Boolean.FALSE.equals(box.getValue())) {
+        if (Boolean.TRUE.equals(allBoxVal) && Boolean.FALSE.equals(box.getValue())) {
           selectAllCheckBox.setValue(false);
-        } else if(Boolean.FALSE.equals(allBoxVal) && areAllSelected()) {
+        } else if (Boolean.FALSE.equals(allBoxVal) && areAllSelected()) {
           selectAllCheckBox.setValue(true);
         }
       }
     });
     /*Necessary to propagate selection of select all checkbox to individual checkbox*/
     selectAllCheckBox.addValueChangeListener(event -> {
-      if(event.isFromClient()) {
+      // needed so boxes that are filtered out are not selected and lead to miscount
```

Comment:
As previously stated, this does not belong here. It is behaviour of the select all checkbox component and not of the individual checkboxes.
This makes it easier to only select attached checkboxes as well.